util/resolver/authorizer.go

```diff
@@ -272,9 +272,14 @@ func (ah *authHandler) doBearerAuth(ctx context.Context) (token string, err erro
 	if r, exist := ah.scopedTokens[scoped]; exist {
 		ah.Unlock()
 		r.Wait()
-		if r.expires.IsZero() || r.expires.After(time.Now()) {
+		if errors.Is(ctx.Err(), context.Canceled) {
```

Comment:
By docs `<-ctx.Done()` should be checked before `Err()`

Reply:
also, this condition only applies if `r.err != nil`

Reply:
Fixed